iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/typed_unique_id.inl

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_INL
+#define IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_INL
+
+namespace iox
+{
+namespace popo
+{
+template <typename T>
+std::atomic<uint64_t> TypedUniqueId<T>::globalIDCounter{0u};
+
+template <typename T>
+inline TypedUniqueId<T>::TypedUniqueId() noexcept
+    : ThisType(cxx::newtype::internal::ProtectedConstructor,
+               (static_cast<uint64_t>(internal::getUniqueRouDiId()) << UNIQUE_ID_BIT_LENGTH)
+                   + ((globalIDCounter.fetch_add(1u, std::memory_order_relaxed) << ROUDI_ID_BIT_LENGTH)
+                      >> ROUDI_ID_BIT_LENGTH))
+{
```

Comment:
The shift back and forth of ROUDI_ID_BIT_LENGTH is to zero the highest bits right? I would have preferred a constexpr mask for both parts (and using & to access the different parts). For positive numbers (i.e. uint) I am sure that it is guaranteed that the upper bits will be zero when we shift right. 

But note that there are some undefined behaviour cases like this which seemingly work, masking always works.